### PR TITLE
docker_api connection plugin

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
This is a revival of ansible/ansible#66114.

Instead of extending the `docker` connection plugin to provide two backends, I added a separate connection plugin `docker_api`.

##### ISSUE TYPE
- New Plugin Pull Request

##### COMPONENT NAME
plugins/connections/docker_api.py

### CODE REVIEW

**Comment by felixfontein** [comment]
ready_for_review

**Comment by felixfontein** [comment]
CC @dustymabe @objectified @ssbarnea since you were interested in this in ansible/ansible#17692.

File: plugins/connection/docker_api.py [hunk 1]
```diff
@@ -0,0 +1,599 @@
...
+    - Run commands or put/fetch files to an existing docker container.
+    - Uses Docker SDK for Python to interact directly with the Docker daemon instead of
+      using the Docker CLI. Use the
+      R(community.docker.docker,ansible_collections.community.docker.docker_connection)
+      connection plugin if you want to use the Docker CLI.
+options:
+    remote_user:
+        type: str
+        description:
+            - The user to execute as inside the container
```
Comment (Andersson007) [inline plugins/connection/docker_api.py:23]:
```suggestion
            - The user to execute as inside the container.
```

**Review by Andersson007 — APPROVED** [review]

**Comment by felixfontein** [comment]
@Andersson007 thanks a lot for reviewing this! I'm happy that this work (which started almost a year ago!) is finally done.

**Pull request merged**